### DISCUSSION TITLE
Z home after every probe?

### DISCUSSION THREAD

**Discussion by KiLLie2k** [Jyers/Marlin#639] [Ideas]
Would it be possible to home z after every probe? I feel like I have some z drift going on that I would like to try to compensate for when creating my mesh... would I have to change the gcode and compile the firmware myself?

**Comment by Tuncay-Ayhan** [comment]
> Would it be possible to home z after every probe? I feel like I have some z drift going on that I would like to try to compensate for when creating my mesh... would I have to change the gcode and compile the firmware myself?

Not sure this is necessary, also this would add a lot of time depending on the config you use. 

In what terms do you see the drift?

**Reply by KiLLie2k** [reply]
I was just reading this write up by someone about mesh leveling and what he is saying made alot of sense to me and i'd like to test his theory by homing z for each point, obviously not with a 15x15 mesh haha

https://jarfil.github.io/blog/2021/04/10/Ender3v2BLTouchFix/

Why it happens
Every point probe actually consists of several moves:

Move XY to probe position
Move Z down until probe touches
Move Z up
Move Z down again, this time slower
Move Z up
That is 4 times that any inaccuracy in the Z axis can introduce an error. For 8x8 = 64 points, that’s 256 moves, which even with 0.001 mm Z axis creep per point, means a 0.256 mm error for the last point, which skews all the measurements.

Effectively, you need less than 0.0001 mm average error per Z move to achieve 0.025 mm precision over 256 moves.

**Reply by projectew** [reply]
My understanding of stepper motors is that while they can have variable accuracies depending on quality/other factors, they are individually highly repeatable in their operation - they should be able to count the steps they're taking while moving down to probe the bed, then move back up the same number of steps, returning to its original Z position in a highly deterministic manner. In the terms you've laid out, I believe that would mean they should have a very, very low drift along any axis so long as it was homed prior to all movements.

**Reply by gothcha** [reply]
I agree @projectew, also the new probe repeatability test will verify this. I get almost zero error when raising, lowering & probing multiple times including the BLTouch itself. I would not worry about this, run the test and you can satisfy yourself.
